editor/src/components/navigator/navigator-item/component-picker.tsx

```diff
@@ -59,20 +59,90 @@ export function componentPickerOptionTestId(componentName: string, variant?: str
 }
 
 export const ComponentPicker = React.memo((props: ComponentPickerProps) => {
+  const { onItemClick } = props
+  const [selectedComponentKey, setSelectedComponentKey] = React.useState<string | null>(null)
   const [filter, setFilter] = React.useState<string>('')
+  const menuRef = React.useRef<HTMLDivElement | null>(null)
 
-  const allComponentsToShow: InsertMenuItemGroup[] = []
-
-  props.allComponents.forEach((c) => {
-    allComponentsToShow.push({
-      ...c,
-      options: c.options.filter((o) =>
-        o.label.toLocaleLowerCase().includes(filter.toLocaleLowerCase().trim()),
-      ),
+  const componentsToShow = useMemo(() => {
```

Comment:
note - remember that for react hooks we conventionally call them as `React.useXXX`